lib/netsuite/records/opportunity.rb

```diff
@@ -0,0 +1,49 @@
+module NetSuite
+  module Records
+    class Opportunity
+      include Support::Fields
+      include Support::RecordRefs
+      include Support::Records
+      include Support::Actions
+      include Namespaces::TranSales
+
+      actions :get, :get_list, :add, :initialize, :delete, :update, :upsert, :search
+
+      fields :bill_address, :title, :estimated_budget, :trans_id, :source,
+        :tran_date, :days_open, :currency_name, :is_budget_approved,
+        :close_date, :last_modified_date, :created_date,
+        :est_gross_profit, :est_gross_profit_percent, :exchange_rate, :expected_close_date,
+        :memo, :probability, :proj_alt_sales_amt, :projected_total,
+        :range_high, :range_low,
+        :ship_is_residential, :ship_address, :source, :status, :sync_partner_teams, :sync_sales_teams,
+        :tax2_total, :tax_total, :total_cost_estimate, :tran_date, :tran_id,
+        :weighted_total
+
+      field :ship_address, ShipAddress
+      field :bill_address, BillAddress
+      field :item_list,                OpportunityItemList
+      field :custom_field_list,        CustomFieldList
+
+      read_only_fields :discount_total, :sub_total, :tax_total, :total
+
+      record_refs :account, :currency, :klass, :department, :entity, :entity_status, :location,
+        :partner, :subsidiary, :sales_group, :sales_rep, :custom_form, :job, :forcast_type,
+        :lead_source, :sales_readiness, :win_loss_reason
+
+      attr_reader :internal_id
+      attr_accessor :external_id
+      attr_accessor :search_joins
+
+      def initialize(attributes = {})
+        @internal_id = attributes.delete(:internal_id) || attributes.delete(:@internal_id)
+        @external_id = attributes.delete(:external_id) || attributes.delete(:@external_id)
+        initialize_from_attributes_hash(attributes)
+      end
+
+      def self.search_class_name
```

Comment:
No need for a `search_Class_name` here... this is auto-generated. This is only required when the search name is different from the class name.